opencensus/stats/measurement_map.py

```diff
@@ -48,6 +57,24 @@ def measure_float_put(self, measure, value):
         """associates the measure of type Float with the given value"""
         self._measurement_map[measure] = value
 
+    def measure_put_attachment(self, key, value):
+        """Associate the contextual information of an Exemplar to this MeasureMap
+            Contextual information is represented as key - value string pairs.
+            If this method is called multiple times with the same key,
+            only the last value will be kept.
+        """
+        if self._attachments is None:
+            raise TypeError('attachments should not be empty')
```

Comment:
If `_attachments` is `None` and users want to put new attachments, it should be created as an empty dict instead of raising error.